documentation/terraform_example.md

```diff
@@ -0,0 +1,448 @@
+private.pkcs7.pem  public.pkcs7.pem
+```
+Encrypt password
+
+```
+$ bolt secret encrypt root
+ENC[PKCS7,MIICeQYJKoZIhvcNAQcDoIICajCCAmYCAQAxggIhMIICHQIBADAFMAACAQEw
+DQYJKoZIhvcNAQEBBQAEggIAY2FIHgScXnMhAf2XZGC1Oae8llfEUIWsurss
+Ygtc5nN1yIJkCY3IU72WhDpNgPeiFThlxXgt5KwhudxAjnisnjB0lRlzzhzX
+heyPh6XFd4EG8AK4ty/s+1bf+TfYq0EY/C6mPbynLYYDXlR+IkCm/5Z5lo0n
+6KQjAmdTUlJx1aCY9lcyK0QZ9erznGFxCyuYwsOGAx+j+52wIoGLp3WS4aHa
+XCKlecD9ylvohK/ZSW2pQcFOF71JHvrJUjnWrQ9ONoIIGrzpUIaTbUDD/w2n
+oVRUFUW/Hx4sMhwU0E0ozTesPcI+yFQPDXhJxvXl15fdzv0hIJ6Moml13MRP
+m3r9U+MK5x9//CGAym7v774/A+8+V97NCW7VW6X1XapIX9RM52FY4ZG3k9sp
+XtnIRGFmNyjz0q1PC1ZuSAulb+RNwLjpVay6RcGydHaIJZ7AqPknA4nSN7Ye
+D49DYl/tSkDdZKU0FXuF0gWBPv52y2t46YnXcBTDArxBqQjRsON+XWyd/9t5
+YYVvWx97oRa6XHdZnRX9KDDXeKWH0e+kzEop6hPJaNOsmSlMJCn3OSyvLJdw
+zYatllKZF+2iSVf2SYTtGqTL4wLs50164+kwUsem1IwlzAgpj0FLw+quQxWX
+CPACM2l0cfIA+fjLxpE7ot5aTymwybtW7EorfmXc5KyfLWIwPAYJKoZIhvcN
+AQcBMB0GCWCGSAFlAwQBKgQQwml96nDZqkQmy4dGIch4NYAQQiu3tflL9OqW
+iV8egTT6Yw==]
+```
+Add to inventory
+```yaml
+version: 2
+groups:
+  - name: terraform-group
+    targets:
+      - _plugin: terraform
+        dir: /home/cas/working_dir/dynamic-inventory-demo
+        resource_type: docker_container.sshd.*
+        # statefile: test/terraform_local.tfstate
+        backend: remote
+        uri: network_data.0.gateway
+        name: name
+        config:
+          ssh:
+            port: ports.0.external
+    config:
+      transport: ssh
+      ssh:
+        user: root
+        password: 
+          _plugin: pkcs7
+          encrypted_value:
+            ENC[PKCS7,MIICeQYJKoZIhvcNAQcDoIICajCCAmYCAQAxggIhMIICHQIBADAFMAACAQEw
+            DQYJKoZIhvcNAQEBBQAEggIAY2FIHgScXnMhAf2XZGC1Oae8llfEUIWsurss
+            Ygtc5nN1yIJkCY3IU72WhDpNgPeiFThlxXgt5KwhudxAjnisnjB0lRlzzhzX
+            heyPh6XFd4EG8AK4ty/s+1bf+TfYq0EY/C6mPbynLYYDXlR+IkCm/5Z5lo0n
+            6KQjAmdTUlJx1aCY9lcyK0QZ9erznGFxCyuYwsOGAx+j+52wIoGLp3WS4aHa
+            XCKlecD9ylvohK/ZSW2pQcFOF71JHvrJUjnWrQ9ONoIIGrzpUIaTbUDD/w2n
+            oVRUFUW/Hx4sMhwU0E0ozTesPcI+yFQPDXhJxvXl15fdzv0hIJ6Moml13MRP
+            m3r9U+MK5x9//CGAym7v774/A+8+V97NCW7VW6X1XapIX9RM52FY4ZG3k9sp
+            XtnIRGFmNyjz0q1PC1ZuSAulb+RNwLjpVay6RcGydHaIJZ7AqPknA4nSN7Ye
+            D49DYl/tSkDdZKU0FXuF0gWBPv52y2t46YnXcBTDArxBqQjRsON+XWyd/9t5
+            YYVvWx97oRa6XHdZnRX9KDDXeKWH0e+kzEop6hPJaNOsmSlMJCn3OSyvLJdw
+            zYatllKZF+2iSVf2SYTtGqTL4wLs50164+kwUsem1IwlzAgpj0FLw+quQxWX
+            CPACM2l0cfIA+fjLxpE7ot5aTymwybtW7EorfmXc5KyfLWIwPAYJKoZIhvcN
+            AQcBMB0GCWCGSAFlAwQBKgQQwml96nDZqkQmy4dGIch4NYAQQiu3tflL9OqW
+            iV8egTT6Yw==]
+```
+Prove it
+```
+$ bolt command run hostname -t terraform-group
+Started on docker_target_0...
+Started on docker_target_2...
+Started on docker_target_1...
+Finished on docker_target_2:
+  STDOUT:
+    268e5d2572d7
+Finished on docker_target_0:
+  STDOUT:
+    f04670444f71
+Finished on docker_target_1:
+  STDOUT:
+    3e32b6d85cfd
```

Comment:
What might be cool is to then `terraform destroy` and `terraform apply` and show that the plugin still works for the newly provisioned nodes.